pkg/registry/core/replicationcontroller/strategy.go

```diff
@@ -47,6 +47,9 @@ import (
 	"sigs.k8s.io/structured-merge-diff/v4/fieldpath"
 )
 
+// Add to package
+var ValidateDeclarativelyFunc = rest.ValidateDeclaratively
```

Comment:
I don't follow - when will this var ever not be this value?  Comment?

Reply:
I don't think we should introduced exported global state in this way to make it possible to mock out validation for testing purposes.

The integration test server is not really designed to mock out behavior like this. 

I do like the signal the integration test provides.  But since a bug free apiserver should never serve mismatches, I don't know of a way to test this behavior from the integration test server.  

It might be better to ship without these tests?  We could look into adding similar tests to a package like staging/src/k8s.io/apiserver/pkg/server but I suspect it is quite a lot of work, and I'm not sure it's worth it.

Reply:
+1, I agree that this is not ideal and for the reasons noted.  I have removed this test (`test/integration/apiserver/declarative_validation_metrics_test.go`) for now and removed the associated ValidateDeclarativelyFunc indirection that was used by the test (now strategy.go uses ValidateDeclaratively directly).